src/frequenz/client/common/metric/__init__.py

```diff
@@ -0,0 +1,141 @@
+    AC_APPARENT_ENERGY_PHASE_3 = PBMetric.METRIC_AC_APPARENT_ENERGY_PHASE_3
+    AC_ACTIVE_ENERGY = PBMetric.METRIC_AC_ACTIVE_ENERGY
+    AC_ACTIVE_ENERGY_PHASE_1 = PBMetric.METRIC_AC_ACTIVE_ENERGY_PHASE_1
+    AC_ACTIVE_ENERGY_PHASE_2 = PBMetric.METRIC_AC_ACTIVE_ENERGY_PHASE_2
+    AC_ACTIVE_ENERGY_PHASE_3 = PBMetric.METRIC_AC_ACTIVE_ENERGY_PHASE_3
+    AC_ACTIVE_ENERGY_CONSUMED = PBMetric.METRIC_AC_ACTIVE_ENERGY_CONSUMED
+    AC_ACTIVE_ENERGY_CONSUMED_PHASE_1 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_CONSUMED_PHASE_1
+    )
+    AC_ACTIVE_ENERGY_CONSUMED_PHASE_2 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_CONSUMED_PHASE_2
+    )
+    AC_ACTIVE_ENERGY_CONSUMED_PHASE_3 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_CONSUMED_PHASE_3
+    )
+    AC_ACTIVE_ENERGY_DELIVERED = PBMetric.METRIC_AC_ACTIVE_ENERGY_DELIVERED
+    AC_ACTIVE_ENERGY_DELIVERED_PHASE_1 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_DELIVERED_PHASE_1
+    )
+    AC_ACTIVE_ENERGY_DELIVERED_PHASE_2 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_DELIVERED_PHASE_2
+    )
+    AC_ACTIVE_ENERGY_DELIVERED_PHASE_3 = (
+        PBMetric.METRIC_AC_ACTIVE_ENERGY_DELIVERED_PHASE_3
+    )
+    AC_REACTIVE_ENERGY = PBMetric.METRIC_AC_REACTIVE_ENERGY
+    AC_REACTIVE_ENERGY_PHASE_1 = PBMetric.METRIC_AC_REACTIVE_ENERGY_PHASE_1
+    AC_REACTIVE_ENERGY_PHASE_2 = PBMetric.METRIC_AC_REACTIVE_ENERGY_PHASE_2
+    AC_REACTIVE_ENERGY_PHASE_3 = PBMetric.METRIC_AC_REACTIVE_ENERGY_PHASE_3
+
+    # AC harmonics
+    AC_THD_CURRENT = PBMetric.METRIC_AC_THD_CURRENT
+    AC_THD_CURRENT_PHASE_1 = PBMetric.METRIC_AC_THD_CURRENT_PHASE_1
+    AC_THD_CURRENT_PHASE_2 = PBMetric.METRIC_AC_THD_CURRENT_PHASE_2
+    AC_THD_CURRENT_PHASE_3 = PBMetric.METRIC_AC_THD_CURRENT_PHASE_3
+
+    # General BMS metrics
+    BATTERY_CAPACITY = PBMetric.METRIC_BATTERY_CAPACITY
+    BATTERY_SOC_PCT = PBMetric.METRIC_BATTERY_SOC_PCT
+    BATTERY_TEMPERATURE = PBMetric.METRIC_BATTERY_TEMPERATURE
+
+    # General inverter metrics
+    INVERTER_TEMPERATURE = PBMetric.METRIC_INVERTER_TEMPERATURE
+    INVERTER_TEMPERATURE_CABINET = PBMetric.METRIC_INVERTER_TEMPERATURE_CABINET
+    INVERTER_TEMPERATURE_HEATSINK = PBMetric.METRIC_INVERTER_TEMPERATURE_HEATSINK
+    INVERTER_TEMPERATURE_TRANSFORMER = PBMetric.METRIC_INVERTER_TEMPERATURE_TRANSFORMER
+
+    # EV charging station metrics
+    EV_CHARGER_TEMPERATURE = PBMetric.METRIC_EV_CHARGER_TEMPERATURE
+
+    # General sensor metrics
+    SENSOR_WIND_SPEED = PBMetric.METRIC_SENSOR_WIND_SPEED
+    SENSOR_WIND_DIRECTION = PBMetric.METRIC_SENSOR_WIND_DIRECTION
+    SENSOR_TEMPERATURE = PBMetric.METRIC_SENSOR_TEMPERATURE
+    SENSOR_RELATIVE_HUMIDITY = PBMetric.METRIC_SENSOR_RELATIVE_HUMIDITY
+    SENSOR_DEW_POINT = PBMetric.METRIC_SENSOR_DEW_POINT
+    SENSOR_AIR_PRESSURE = PBMetric.METRIC_SENSOR_AIR_PRESSURE
+    SENSOR_IRRADIANCE = PBMetric.METRIC_SENSOR_IRRADIANCE
+
+    @classmethod
+    def from_proto(cls, metric: PBMetric.ValueType) -> Metric:
```

Comment:
```suggestion
    def from_proto(cls, metric: PBMetric.ValueType) -> Self:
```

Reply:
I actually get a `mypy` error from using `Self` here > not sure how to change that now?

Reply:
I get a mypy error if I use `Self` instead of `Metric`.

Reply:
Discussed with @Marenz and resolved.